Mongo.Migration/Services/MongoDB/MongoRegistrator.cs

```diff
@@ -21,16 +21,21 @@ public MongoRegistrator(DocumentVersionSerializer serializer, MigrationIntercept
 
         public void Register()
         {
+            RegisterSerializationProvider();
+            RegisterSerializer();
+        }
+
+        private void RegisterSerializationProvider()
+        {
             BsonSerializer.RegisterSerializationProvider(_provider);
+        }
 
-            try
-            {
-                BsonSerializer.RegisterSerializer(typeof(DocumentVersion), _serializer);
-            }
-            catch (BsonSerializationException Exception)
-            {
-                // Catch if Serializer was registered already, not great. But for testing it must be catched.
-            }
+        private void RegisterSerializer()
+        {
+            var foundSerializer = BsonSerializer.LookupSerializer(_serializer.ValueType);
```

Comment:
That does not work, the BsonSerializer has always a default "DocumentVersionSerializer". That is why this ugly try/catch is here :)